Lib/shutil.py

```diff
@@ -224,18 +240,19 @@ def copyfile(src, dst, *, follow_symlinks=True):
     file_size = 0
     for i, fn in enumerate([src, dst]):
         try:
-            st = os.stat(fn)
+            st = _stat(fn)
```

Comment:
On Windows, DirEntry.stat(follow_symlinks=False) doesn't need a syscall. Maybe it would be worth it to try to pass follow_symlinks here? Or try to call stat() with follow_symlinks=False to manually do a second stat call if it's a symlink and we want to follow symlinks?

Reply:
Mmm... honestly I'm not sure. It looks like this should (may) have been done when `follow_symlinks` was introduced in Python 3.3 but it wasn't. I'm not sure about the repercussions of changing it in here. I'd rather do it as part of another BPO / PR.